src/network/networkd-ndisc.c

```diff
@@ -214,7 +286,38 @@ static int ndisc_router_process_autonomous_prefix(Link *link, sd_ndisc_router *r
 
         if (in_addr_is_null(AF_INET6, (const union in_addr_union *) &link->network->ipv6_token) == 0)
                 memcpy(((char *)&address->in_addr.in6) + 8, ((char *)&link->network->ipv6_token) + 8, 8);
-        else {
+        else if (link->network->ipv6_address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE) {
+                struct in6_addr addr = address->in_addr.in6;
+                IPv6Token *i;
+
+                LIST_FOREACH(tokens, i, link->network->ipv6tokens) {
+                        if (!memcmp(&i->prefix, &addr, FAMILY_ADDRESS_SIZE(address->family))) {
+
+                                for (; i->dad_counter < DAD_CONFLICTS_IDGEN_RETRIES_RFC7217; i->dad_counter++) {
```

Comment:
When the counter is reset?

Reply:
https://tools.ietf.org/html/rfc7217#page-12
6.  Resolving DAD Conflicts
```
In the event that a DAD conflict cannot be solved (possibly after
 trying a number of different addresses), address configuration would
 fail.  In those scenarios, hosts MUST NOT automatically fall back to
 employing other algorithms for generating Interface Identifiers.
```

I think this is not meant to be reset. It should fail. If the link goes down/up or networkd gets restarted then it will start again.

Reply:
Hmm... What happens when networkd runs so long?

Reply:
The address has an expired time when it gets removed then again starts from beginning. Hence should not be a problem. dad_counter is inside the Token object.

Reply:
Hmm. The sentence you quoted does not mean we should _not_ reset the counter...

Reply:
it says `address configuration would fail. and must not fallback` If we reset then it would keep on trying again which it should not.